typescript/sdk/src/token/adapters/EvmTokenAdapter.ts

```diff
@@ -192,6 +200,85 @@ export class EvmTokenAdapter<T extends ERC20 = ERC20>
   }
 }
 
+// Interacts with LSP7 contracts
+export class EvmLSP7TokenAdapter<
+    T extends LSP7DigitalAssetInitAbstract = LSP7DigitalAssetInitAbstract,
+  >
+  extends EvmNativeTokenAdapter
+  implements ITokenAdapter<PopulatedTransaction>
+{
+  public readonly contract: T;
+
+  constructor(
+    public readonly chainName: ChainName,
+    public readonly multiProvider: MultiProtocolProvider,
+    public readonly addresses: { token: Address },
+    public readonly contractFactory: any = LSP7DigitalAssetInitAbstract__factory,
+  ) {
+    super(chainName, multiProvider, addresses);
+    this.contract = contractFactory.connect(
+      addresses.token,
+      this.getProvider(),
+    );
+  }
+
+  override async getBalance(address: Address): Promise<bigint> {
+    const balance = await this.contract.balanceOf(address);
+    return BigInt(balance.toString());
+  }
+
+  override async getMetadata(isNft?: boolean): Promise<TokenMetadata> {
+    const [decimals, symbolHex, nameHex] = await Promise.all([
+      isNft ? 0 : this.contract.decimals(),
+      this.contract['getData(bytes32)'](LSP4DataKeys.LSP4TokenSymbol),
+      this.contract['getData(bytes32)'](LSP4DataKeys.LSP4TokenName),
+    ]);
+
+    const symbol = toUtf8String(symbolHex);
+    const name = toUtf8String(nameHex);
+
+    return { decimals, symbol, name };
+  }
+
+  override async isApproveRequired(
+    owner: Address,
+    spender: Address,
+    weiAmountOrId: Numberish,
+  ): Promise<boolean> {
+    const allowance = await this.contract.authorizedAmountFor(spender, owner);
+    return allowance.lt(weiAmountOrId);
+  }
+
+  override populateApproveTx({
+    weiAmountOrId,
+    recipient,
+  }: TransferParams): Promise<PopulatedTransaction> {
+    return this.contract.populateTransaction[
+      'authorizeOperator(address,uint256,bytes)'
+    ](recipient, weiAmountOrId.toString(), '0x');
+  }
+
+  override populateTransferTx({
+    fromAccountOwner: from,
+    recipient,
+    weiAmountOrId,
+  }: TransferParams): Promise<PopulatedTransaction> {
+    if (!from) throw new Error('fromAccountOwner required for LSP7');
+    return this.contract.populateTransaction.transfer(
+      from,
+      recipient,
+      weiAmountOrId.toString(),
+      true,
```

Comment:
Suggest adding small comment

```suggestion
      true, // `force` parameter set to true to allow transfers to any addresses
```

we could also add a small reference to the LSP standard document in comment